swabbie-core/src/test/kotlin/com/netflix/spinnaker/swabbie/work/WorkProcessorTest.kt

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.work
+
+import com.netflix.spectator.api.NoopRegistry
+import com.netflix.spinnaker.swabbie.LockingService
+import com.netflix.spinnaker.swabbie.ResourceTypeHandler
+import com.netflix.spinnaker.swabbie.events.Action
+import com.netflix.spinnaker.swabbie.model.AWS
+import com.netflix.spinnaker.swabbie.model.SpinnakerAccount
+import com.netflix.spinnaker.swabbie.model.WorkConfiguration
+import com.netflix.spinnaker.swabbie.test.InMemoryWorkQueue
+import com.netflix.spinnaker.swabbie.test.TestResource
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.doReturn
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import org.junit.jupiter.api.Test
+import java.time.Clock
+
+object WorkProcessorTest {
+  @Test
+  fun `should do nothing if work queue is empty`() {
+    val workQueue = InMemoryWorkQueue()
+    val handler = mock<ResourceTypeHandler<TestResource>>()
+
+    whenever(handler.handles(any())) doReturn true
+
+    val processor = WorkProcessor(
```

Comment:
You define this for every test in here. Can we have this in either a common function, or just as the `@Subject` (or the equivalent - i can't remember if thats the right syntax for this test framework, it's probably not) of the class?